bike_kollective/lib/add_bike_screen.dart

```diff
@@ -1,80 +1,96 @@
-import 'package:cloud_firestore/cloud_firestore.dart';
+import 'package:bike_kollective/data/provider/owned_bikes.dart';
+import 'package:bike_kollective/data/model/bk_document_reference.dart';
+import 'package:bike_kollective/data/model/bk_geo_point.dart';
+import 'package:bike_kollective/data/model/bike.dart';
+import 'package:image_picker/image_picker.dart';
+import 'dart:io';
+import 'package:flutter/foundation.dart' show kIsWeb;
 
-// Bike model
-class Bike {
-  final String name;
-  final String type;
-  final String description;
-  final String code;
-
-  Bike({
-    required this.name,
-    required this.type,
-    required this.description,
-    required this.code,
-  });
-}
-
-// State management class using ChangeNotifier
 class BikeNotifier extends ChangeNotifier {
-  String? _name;
-  String? _type;
-  String? _description;
-  String? _code;
-
-  List<String> bikeTypes = ['Road Bike', 'Mountain Bike', 'E-bike'];
+  late TextEditingController nameController;
+  late TextEditingController descriptionController; // Add separate controller
+  late TextEditingController lockCodeController; // Add separate controller
+  BikeType? _type;
+  File? _image; // Change this to the notifier to retain state
 
-  void setName(String name) {
-    _name = name;
-    notifyListeners();
+  BikeNotifier() {
+    nameController = TextEditingController();
+    descriptionController = TextEditingController(); // Initialize the description controller
+    lockCodeController = TextEditingController(); // Initialize the lock code controller
   }
 
-  void setType(String type) {
+  void setType(BikeType type) {
     _type = type;
     notifyListeners();
   }
 
   void setDescription(String description) {
-    _description = description;
+    descriptionController.text = description; // Update the description controller
     notifyListeners();
   }
 
   void setLockCode(String lockCode) {
-    _code = lockCode;
+    lockCodeController.text = lockCode; // Update the lock code controller
     notifyListeners();
   }
 
-  Bike? get bike {
-    if (_name != null && _type != null && _description != null && _code != null) {
-      return Bike(name: _name!, type: _type!, description: _description!, code: _code!);
+  void setImage(File image) {
+    _image = image; // Set selected image
+    notifyListeners(); // Notify listeners to update UI
+  }
+
+  BikeModel? get bike {
+    if (nameController.text.isNotEmpty && _type != null && descriptionController.text.isNotEmpty && lockCodeController.text.isNotEmpty) {
+      return BikeModel.newBike(
+        owner: BKDocumentReference.fake("ownerRef"),
+        name: nameController.text,
+        type: _type!,
+        description: descriptionController.text, // Get description from its controller
+        code: lockCodeController.text, // Get lock code from its controller
+        imageLocalPath: '', // Use the imageLocalPath here if applicable
+        startingPoint: BKGeoPoint(0, 0),
```

Comment:
Use the user location provider data.

Reply:
Comment about this being a temporary placeholder.

Reply:
You might consider a small class of your own for this, like BikeFormData. Then you can just have what you need. But this is ok given running out of time.